src/Api/referrals.js

```diff
@@ -342,34 +317,45 @@ export async function getReferralCodeOwner(chainId, referralCode) {
   const referralStorageAddress = getContract(chainId, "ReferralStorage");
   const provider = getProvider(null, chainId);
   const contract = new ethers.Contract(referralStorageAddress, ReferralStorage.abi, provider);
-  const codeOwner = await contract.codeOwners(referralCode);
-  return codeOwner;
+  try {
```

Comment:
no need to change it in this PR, but i think in future would be good to use `useSWR` for all data retrievals

for example useSWR has retries out-of-the-box. imagine in this specific case something goes wrong, request fails and it will be never requested again. and we will display in UI that code has no owner whereas it might be not true

Reply:
Sure!

Reply:
I will add a task for that and update it later.